### DISCUSSION TITLE
MUSHR for racing

### DISCUSSION THREAD

**Discussion by abhisheksreesaila** [prl-mushr/mushr#52] [General]
I know we have a tutorial for implementing DL using a modified version of a donkey simulator.  I want to do the same thing but on the car and use it in upcoming races.  
 - I assume we can follow the same steps as donkey car? Will I need to watch out for any specific thing?
 - I see MUSHR SIM as a modified version. Was there any major modification?

**Comment by sidtalia** [comment]
Hi Abhishek,

The DL implementation is merely a way for new MuSHR enthusiasts to dabble into the use of DL for autonomous driving. While the model can be trained in simulation, the framework itself is not connected to ROS and thus will not work directly on the physical car. If you wish to implement the DL system on the real MuSHR car, you can use the DL implementation framework as inspiration. 

Further, I would advise using real-world data for training purposes, as the sim to real transfer of the DL framework has not been verified by us.

As for your second question, yes, the MuSHR DL sim is a modified version of the donkey sim. The major modifications are:
1) 3 cameras instead of 1, facing left, right and center. This was done to help produce a more balanced dataset and avoid covariate shift.
2) New, simpler environments that are useful for benchmarking performance.

Regards,
Sidharth.

**Reply by abhisheksreesaila** [reply]
Thanks for the prompt response.  Just for racing purposes, I will use a new sd card and deploy donkey env and use your code as a basis for training/testing the model. 
- Do you have the source code for the modifications to the donkey sim? Are you guys planning to make it open source.  I am planning to install stock donkey sim first, train/test and if time permits,  edit that to have 3 cameras as per your recommendation.

**Comment by sidtalia** [comment]
Hi, here's the [code](https://github.com/naughtyStark/mushr-sdsandbox) for the simulator (you will need unity to modify this)
